### ISSUE TITLE
Question: serving all assets on /static

### ISSUE THREAD

**Issue by jsfr** [davezuko/react-redux-starter-kit#766]
I would like to serve all the assets generated by webpack (either with HMR and dev-middleware or when compiling) from `/static/*` instead of the root `/` as is currently done (the reasoning behind this is, to also serve an API on `/api/` and be sure to avoid name conflicts).

How would I go about doing this? I tried changing the `publicPath` for webpack, which results in it requesting the files from there. However this just gives 404's. It seems this way of doing it, results in `index.html` also being served on `/static/index.html` instead of root, which is of course not what I want.

**TL;DR** How do I serve all webpack generated assets on `/static/` correctly and still serve `index.html` on `/`?

Thanks in advance :)

**Comment by davezuko** [comment]
Not sure off the top of my head, sorry. I will give it a shot when I get a chance, but let me know if you make any progress on your own.

**Comment by fixpunkt** [comment]
As a workaround, you can set `publicPath` to `/static` and change the configuration of the `historyApiFallback` middleware in `server/main.js` like so:

    do app.use(convert(historyApiFallback({
      verbose: false,
      index: '/static/index.html'
    })))

**Comment by fixpunkt** [comment]
The advice I gave in the previous comment was partially wrong. What worked for me:

1. Set `compiler_public_path` to `/static/` in `config/_base.js`.
2. Remove `compiler_public_path` from all other files in `config/`.
3. Configure `historyApiFallback` as described above
4. Change the statement that adds `webpackHMRMiddleware` in `server/main.js` like this:
`app.use(webpackHMRMiddleware(compiler, {
        path: '/__webpack_hmr'
      }))`
5. Change line 30 (the dev-environment entry config) in `build/webpack.config.js` like this:
`    ? [APP_ENTRY_PATH, 'webpack-hot-middleware/client?path=/__webpack_hmr']`

I hope this is reasonably clear.

**Comment by fixpunkt** [comment]
On a related note, the HMR endpoint should probably not depend on `compiler_public_path`, or am I missing something here?

**Comment by davezuko** [comment]
Thanks for taking the time to answer the question @fixpunkt , I really appreciate it. Going to close this as resolved unless somebody lets me know otherwise.

As far as your second question, I posted a reply to that in [this issue](https://github.com/davezuko/react-redux-starter-kit/issues/757#issuecomment-217715911). Perhaps I'm wrong, in which case we can eliminate some of these headaches.

**Comment by jonricaurte** [comment]
@davezuko I think there should be two paths in the config. One should be for publicPath and the other should be for appPath. Something like below:

`
// webpack.config.js
webpackConfig.entry = {
   app : __DEV__
    ? [APP_ENTRY].concat(`webpack-hot-middleware/client?path=${config.compiler_app_path}__webpack_hmr`)
     : [APP_ENTRY],
   vendor : config.compiler_vendors
 };
// environment.js
development : config => ({
    compiler_public_path : `http://${config.server_host}:${config.server_port}/static/`,
    compiler_app_path    : `http://${config.server_host}:${config.server_port}/`
}),
// server/main.js
app.use(require('connect-history-api-fallback')({
  verbose: false,
  index: '/static/index.html'
}));
app.use(require('webpack-hot-middleware')(compiler, {
   log: console.log, // eslint-disable-line no-console
   path: '/__webpack_hmr',
   heartbeat: 10 * 1000,
}));
`

I think more people are likely to have an assets path and it took me a few hours to figure out what was going wrong until I landed on this issue.

What do you think?
